fpm/src/fpm_command_line.f90

```diff
@@ -16,87 +23,339 @@ module fpm_command_line
     end type
 
-    type, extends(fpm_cmd_settings) :: fpm_new_settings
+    integer,parameter :: ibug=4096
+    type, extends(fpm_cmd_settings)  :: fpm_new_settings
+        character(len=:),allocatable :: name
+        logical                      :: with_executable=.false.
+        logical                      :: with_test=.false.
     end type
 
-    type, extends(fpm_cmd_settings) :: fpm_build_settings
+    type, extends(fpm_cmd_settings)  :: fpm_build_settings
+        logical                      :: release=.false.
     end type
 
-    type, extends(fpm_cmd_settings) :: fpm_run_settings
+    type, extends(fpm_cmd_settings)  :: fpm_run_settings
+        !!character(len=:),allocatable :: name(:)
+        character(len=ibug),allocatable :: name(:)
+        logical                      :: release=.false.
+        character(len=:),allocatable :: args
     end type
 
-    type, extends(fpm_cmd_settings) :: fpm_test_settings
+    type, extends(fpm_cmd_settings)  :: fpm_test_settings
+        !!character(len=:),allocatable :: name(:)
+        character(len=ibug),allocatable :: name(:)
+        logical                      :: release=.false.
+        character(len=:),allocatable :: args
     end type
 
-    type, extends(fpm_cmd_settings) :: fpm_install_settings
+    type, extends(fpm_cmd_settings)  :: fpm_install_settings
     end type
 
+    character(len=:),allocatable :: name
+    !!character(len=:),allocatable :: names(:)
+    character(len=ibug),allocatable :: names(:)
+
 contains
     subroutine get_command_line_settings(cmd_settings)
         class(fpm_cmd_settings), allocatable, intent(out) :: cmd_settings
 
-        character(len=100) :: cmdarg
-
-        if (command_argument_count() == 0) then
-            call print_help()
-        else if (command_argument_count() == 1) then
-            call get_command_argument(1, cmdarg)
-            select case(trim(cmdarg))
-                case("new")
-                    allocate(fpm_new_settings :: cmd_settings)
-                case("build")
-                    allocate(fpm_build_settings :: cmd_settings)
-                case("run")
-                    allocate(fpm_run_settings :: cmd_settings)
-                case("test")
-                    allocate(fpm_test_settings :: cmd_settings)
-                case("install")
-                    allocate(fpm_install_settings :: cmd_settings)
-                case default
-                    print *, "fpm error: No such command " // trim(cmdarg)
-                    error stop 1
-            end select
-        else
-            print *, "Too many arguments"
-            error stop 1
-        end if
-    end subroutine
+        character(len=4096)           :: cmdarg
+        integer                       :: i
+        character(len=:), allocatable :: help_text(:), version_text(:)
+
+        ! text for --version switch,
+        version_text = [character(len=80) :: &
+           &  'VERSION:     0.0.0, 20200920',                               &
```

Comment:
```suggestion
           &  'VERSION:     0.0.0, Pre-alpha',                               &
```
This may be better to avoid workload of updating the date at each pull request.

Reply:
This is actually conflicting with the version specified in the `fpm.toml`, which says it is 0.1.0:
https://github.com/fortran-lang/fpm/blob/973a2cb5bd25f06f1c308c655546a9021667e3e1/fpm/fpm.toml#L1-L3

Reply:
Good spot @awvwgk . @urbanjost, would you mind matching `fpm.toml` here with instead:
```
'VERSION:     0.1.0, Pre-alpha'
```